state/trackableDataTrie.go

```diff
@@ -110,10 +123,91 @@ func (tdaw *trackableDataTrie) SaveKeyValue(key []byte, value []byte) error {
 		return data.ErrLeafSizeTooBig
 	}
 
-	tdaw.dirtyData[string(key)] = value
+	dataEntry := dirtyData{
```

Comment:
just one idea, not related to the code here, maybe we can avoid re-computing the data trie roothash (and intermediate hashes) after every transaction execution, and do it like for the main trie on commit.

This could be done on another PR and benchmark it.

Reply:
Added TODO.